vive/source/viveGlove.cpp

```diff
@@ -8,20 +8,351 @@
 // The above copyright notice and this permission notice shall be included in all copies of the software.
 ================================================================= */
 
-
 #include "mujoco.h"
-#include "stdio.h"
 #include "stdlib.h"
 #include <string>
+#include <stdio.h>
 
 #include "GL/glew.h"
 #include "glfw3.h"
-#include <windows.h>
 #include <openvr.h>
 using namespace vr;
 
-#include "cyberGlove_utils.h"	// cyberGlove
-cgOption* opt;					// cyber glove options
+#if defined(_WIN32)
+#  include <windows.h>
+#  include "CyberGlove_utils.h"	// cyberGlove
+cgOption* opt;
+#elif defined(__unix__)
+#include <cstring>
+//using namespace std;
+#include <unistd.h>
+#include <math.h>
+#include <stdlib.h>
+typedef struct _options
```

Comment:
Can you help me understand what's going on here? It largely looks like a copy-paste of definitions from the header file. @Kelym

Reply:
The Linux build does not support CyberGlove (build error) and cannot import CyberGlove header files. Currently, all options to the puppet applications are stored in `CyberGlove_utils.h` `CyberGlove_utils.cpp`, even the options that have nothing to do with CyberGlove.
Therefore, the Linux build has to duplicate the CyberGloveOption class (and a few more utility functions).